protocols/imap/src/main/java/org/apache/james/imap/message/response/LSubResponse.java

```diff
@@ -20,12 +20,21 @@
 
 import org.apache.james.imap.api.message.response.ImapResponseMessage;
 import org.apache.james.imap.api.process.MailboxType;
+import org.apache.james.mailbox.model.MailboxMetaData;
 
 /**
  * Values an IMAP4rev1 <code>LIST</code> response.
  */
 public final class LSubResponse extends AbstractListingResponse implements ImapResponseMessage {
+
+    private static MailboxMetaData.Selectability fromNoSelect(boolean noSelect) {
+        if (noSelect) {
+            return MailboxMetaData.Selectability.NOSELECT;
+        }
+        return MailboxMetaData.Selectability.NONE;
+    }
+
     public LSubResponse(String name, boolean noSelect, char delimiter) {
```

Comment:
why you don't get rid of this boolean and use directly MailboxMetaData.Selectability here ?

Reply:
Because in LSub you never have Mailbox metadata. You just have "mailbox name" as RFC-3501 is never enforcing registered mailboxes existance.

Reply:
I don't understand the answer. I asked to change from `boolean noSelect` to `MailboxMetaData.Selectability selectability`

Reply:
I didn't solved that problem and decided to limit the scope of the refactoring for now.
It's not clear to me why we restrict flags in LSUB response.